packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -245,41 +260,293 @@ class FilteringTextInputFormatter extends TextInputFormatter {
   /// two "o"s replaced by a single occurrence of the replacement
   /// string) because both of the "o"s would be matched simultaneously
   /// by the pattern.
-  ///
-  /// Additionally, each segment of the string before, during, and
-  /// after the current selection in the [TextEditingValue] is handled
-  /// separately. This means that, in the case of the "Into the Woods"
-  /// example above, if the selection ended between the two "o"s in
-  /// "Woods", even if the pattern was `RegExp('o+')`, the result
-  /// would be "Int* the W**ds", since the two "o"s would be handled
-  /// in separate passes.
-  ///
-  /// See also [String.splitMapJoin], which is used to implement this
-  /// behavior in both cases.
   final String replacementString;
 
+  /// Determines how to perform formatting, update the selection and the
+  /// composing region, on each banned pattern in the input string.
```

Comment:
```suggestion
  /// composing region, on each region to be replaced in the input string.
```

Reply:
I actually used "region to be replaced" here but then realized the existing documentation uses "banned pattern" instead. I think I still prefer "banned pattern" as it conveys the idea better (the user needs to provide a `filterPattern` so "banned pattern" is relatable), also the region may not get replaced (for instance, if it's inside a composing region in the other strategy).